setup.cfg

```diff
@@ -11,3 +11,43 @@ cover-erase=1
 [aliases]
 dev = develop easy_install zodburi[testing]
 docs = develop easy_install zodburi[docs]
+
+[tox:tox]
+envlist =
+    {py27,py34,py35,py36,pypy,pypy3}-{zodb4,zodb5},py37-zodb5,cover,docs
+
+[testenv]
+commands =
+    python setup.py -q test -q
+deps =
+    mock
+    zodb4: ZODB==4.*
+    zodb5: ZODB==5.*
+    zodb4: ZEO==4.*
+    zodb5: ZEO==5.*
+
+[testenv:cover]
+basepython =
+    python2.7
+commands =
+    python setup.py nosetests --with-xunit --with-xcoverage
+deps =
+    {[testenv]deps}
+    nose
+    coverage
+    nosexcover
+
+# we separate coverage into its own testenv because a) "last run wins" wrt
+# cobertura jenkins reporting and b) pypy and jython can't handle any
+# combination of versions of coverage and nosexcover that i can find.
+# coverage==3.4 is required by nosexcover.
+
+
+[testenv:docs]
+basepython =
+    python2.7
+commands =
+    sphinx-build -b html -d docs/_build/doctrees docs docs/_build/html
+deps =
+    Sphinx
+    pylons-sphinx-themes
```

Comment:
Please put this content back in `tox.ini` --  moving the project into "one config file to rule them all" mode is a separate, and more controversial change than adding testing support for `pypy` / `pypy3`.

Reply:
See how this was handled in Pyramid. https://github.com/Pylons/pyramid/commit/9c2373c3b32f587061eae387edf2678a4307897a#diff-b91f3d5bd63fcd17221b267e851608e8

Reply:
Ok I removed the commit! Sorry for that.